src/Illuminate/Events/Dispatcher.php

```diff
@@ -357,7 +360,7 @@ protected function addInterfaceListeners($eventName, array $listeners = [])
     {
         foreach (class_implements($eventName) as $interface) {
             if (isset($this->listeners[$interface])) {
-                foreach ($this->listeners[$interface] as $names) {
+                foreach ($this->getListeners($interface) as $names) {
```

Comment:
Odd to call this after doing an isset on the property just above.

Reply:
without this change, the tests won't pass.
it because `$this->listeners` are now actually "Raw Listeners" and get listeners somewhat bakes them into closures.

Reply:
This part got simplified. `prepareListeners` is now called instead of `getListeners`